src/drivers/intel/cavs/timer.c

```diff
@@ -115,36 +120,52 @@ static int platform_timer_register(struct timer *timer,
 
 int timer_register(struct timer *timer, void(*handler)(void *arg), void *arg)
 {
+	unsigned int cpu = cpu_get_id();
+	unsigned int i;
 	int ret;
 
 	switch (timer->id) {
 	case TIMER0:
 	case TIMER1:
 	case TIMER2:
 		ret = arch_timer_register(timer, handler, arg);
+		if (ret < 0)
+			return ret;
 		/*
 		 * Actually this isn't needed for arch_interrupt_register(),
 		 * since arch_interrupt_unregister() doesn't support interrupt
 		 * sharing and thus doesn't need the handler argument to locate
 		 * the handler, do it just for uniformity
 		 */
-		if (!ret)
-			timer->irq_arg = arg;
-		return ret;
+		timer->irq_arg = arg;
+		break;
 	case TIMER3:
-		return platform_timer_register(timer, handler, arg);
+		ret = platform_timer_register(timer, handler, arg);
+		if (ret < 0)
+			return ret;
+		break;
 	default:
 		return -EINVAL;
 	}
+
+	for (i = 0; i < PLATFORM_CORE_COUNT; i++) {
+		if (i == cpu)
+			continue;
+
+		dcache_invalidate_region(platform_generic_queue + i,
```

Comment:
I think it still doesn't resolve our issue with not up to date timer->logical_irq.
Master core will register timer interrupt first, then after some time slave core will do it.
Master core still won't have invalidated data.